lib/src/main/java/ua/naiksoftware/stomp/StompClient.java

```diff
@@ -267,14 +268,20 @@ private Completable subscribePath(String destinationPath, @Nullable List<StompHe
         headers.add(new StompHeader(StompHeader.ACK, DEFAULT_ACK));
         if (headerList != null) headers.addAll(headerList);
         return send(new StompMessage(StompCommand.SUBSCRIBE,
-                headers, null));
+                headers, null))
+                .doOnError(throwable -> unsubscribePath(destinationPath).subscribe());
```

Comment:
Add an error handler on the send to unsubscribe if the initial Subscribe frame does not work correctly.